pageserver/src/config.rs

```diff
@@ -118,6 +127,53 @@ pub struct PageServerConf {
     pub remote_storage_config: Option<RemoteStorageConfig>,
 }
 
+pub const TENANT_CONFIG_NAME: &str = "config";
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TenantConf {
+    pub checkpoint_distance: u64,
+    pub checkpoint_period: Duration,
+    pub gc_horizon: u64,
+    pub gc_period: Duration,
+    pub pitr_interval: Duration,
+}
+
+impl TenantConf {
+    pub fn from(conf: &PageServerConf) -> TenantConf {
+        TenantConf {
+            gc_period: conf.gc_period,
+            gc_horizon: conf.gc_horizon,
+            pitr_interval: conf.pitr_interval,
+            checkpoint_distance: conf.checkpoint_distance,
+            checkpoint_period: conf.checkpoint_period,
+        }
+    }
+
+    pub fn save(&self, conf: &'static PageServerConf, tenantid: ZTenantId) -> Result<()> {
+        let _enter = info_span!("saving tenant config").entered();
+        let path = conf.tenant_path(&tenantid).join(TENANT_CONFIG_NAME);
+        let mut file =
+            VirtualFile::open_with_options(&path, OpenOptions::new().write(true).create_new(true))?;
+        let config_bytes = self.ser()?;
+        if file.write(&config_bytes)? != config_bytes.len() {
+            bail!("Could not write all the metadata bytes in a single call");
+        }
+        file.sync_all()?;
```

Comment:
Should we do proper dance with fsyncing the directory on first save etc? for reference you can see how metadata file is written to disk

Reply:
I thought that config is less critical than metadata so loosing it is not fatal.
But at the same time it is rarely updated, so not a problem ti fsync parent dir as well.